source/dub/internal/utils.d

```diff
@@ -260,6 +260,9 @@ void download(string url, string filename, uint timeout = 8, string safeUrl = nu
 /// ditto
 void download(URL url, NativePath filename, uint timeout = 8)
 {
+	version (Have_vibe_d_http) {
+		assert(url.username.length == 0 && url.password.length == 0, "Auth not supported yet.");
+	}
```

Comment:
I have been thinking about if this assert is a good idea - I think it would actually help us keep the docs up-to-date and also allow to write a changelog, once the vibe.d urltransfer module supports HTTP auth. However I think there should be a note here, that this restriction comes from vibe.d not supporting it initially and to remove it (and the docs) once it does

Reply:
Is the reason the same for timeout? Vibe-d doesn't support it?

Reply:
I think that's gonna be the case here, but didn't check source